### DISCUSSION TITLE
Is there any way to pass in some sort of context (for example, a logger) to the hooks?

### DISCUSSION THREAD

**Discussion by coleabbeduto-NM** [sindresorhus/got#1544] [Q&A] [answered]
#### What would you like to discuss?
I looked at the documentation as well as the open and closed issues and I didn't see anything like what I would like to do. 

I would love to be able to pass in context on a per request basis into the hooks (specifically beforeRetry and afterResponse) so that we could give it our logger that has all of our express request properties so we can have standardized logging of external requests and retires. 

Is this possible with the current implementation and I've just missed it? Or if it is not currently possible, is this something that has ever been considered/would be considered for the future? 
...

#### Checklist

- [x] I have read the documentation.
- [x] I have looked through existing issues.

**Comment by szmarczak** [comment]
It is currently possible, but there's a plan to unify all those events so logging would be much simpler. See #559

**Comment by coleabbeduto-NM** [comment]
Hey @szmarczak, I appreciate the link to the issue where this may be fixed later on. But how would I go about fixing it with the current implementation (_"It is currently possible"_)?

**Comment by szmarczak** [comment, marked as answer]
You can use `got.extend`. Pass `{context: {logger}}` as the options and attach some hooks e.g. beforeRequest, beforeRetry and beforeError. Then use the options argument or an error object to extract the data you want to log.
